frontend/src/general/LogoutMenu/__tests__/LogoutMenu.test.jsx

```diff
@@ -0,0 +1,197 @@
+import React from "react";
+import Enzyme from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+import { beforeEach, describe, expect, it, jest } from "@jest/globals";
+
+import {
+  ClickAwayListener,
+  Grow,
+  MenuItem,
+  MenuList,
+  Paper,
+  Popper,
+  Typography
+} from "@material-ui/core";
+import { NavLink } from "react-router-dom";
+import LogoutMenu from "../LogoutMenu";
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe("<LogoutMenu/> functional component", () => {
+  let wrapper;
+  const dummyAnchor = { thing: "thing" };
+  const dummyOpen = true;
+  let mockHandleClose;
+  let mockHandleLogout;
+
+  describe("returns a component that", () => {
+    beforeEach(() => {
+      mockHandleClose = jest.fn();
+      mockHandleLogout = jest.fn();
+      wrapper = Enzyme.shallow(
+        <LogoutMenu
+          anchor={dummyAnchor}
+          open={dummyOpen}
+          handleClose={mockHandleClose}
+          handleLogout={mockHandleLogout}
+        />
+      );
+    });
+    it("Contains one <Popper/> component with expected props", () => {
+      const components = wrapper.find(Popper);
+      expect(components).toHaveLength(1);
+
+      const props = components.at(0).props();
+      const expected = {
+        open: dummyOpen,
+        anchorEl: dummyAnchor,
+        transition: true,
+        disablePortal: true
+      };
+      expect(props).toMatchObject(expected);
+    });
+    it("Contains one <Grow/> component with expected props", () => {
+      const components = wrapper.find(Grow);
+      expect(components).toHaveLength(1);
+
+      const props = components.at(0).props();
+      const expected = {
+        in: dummyOpen,
+        style: { transformOrigin: "center top" }
+      };
+      expect(props).toMatchObject(expected);
+    });
+    it("Contains one <Paper/> component with expected props", () => {
+      const components = wrapper.find(Paper);
+      expect(components).toHaveLength(1);
+    });
+    it("Contains one <ClickAwayListener/> component with expected props", () => {
+      const components = wrapper.find(ClickAwayListener);
+      expect(components).toHaveLength(1);
+
+      const props = components.at(0).props();
+      const expected = {
+        onClickAway: mockHandleClose
+      };
+      expect(props).toMatchObject(expected);
+    });
+    it("Contains one <MenuList/> component with expected props", () => {
+      const components = wrapper.find(MenuList);
+      expect(components).toHaveLength(1);
+
+      const props = components.at(0).props();
+      const expected = {
+        autoFocusItem: dummyOpen,
+        id: "menu-list-grow"
+      };
+      expect(props).toMatchObject(expected);
+    });
+    it("Contains 4 <MenuItem/> components with expected props", () => {
```

Comment:
Would it be better to have a describe block in here that focuses just on `MenuItems` for readability? It's minor, if it's not worth adding then that's okay too.

Reply:
Nope